code/__HELPERS/reagents.dm

```diff
@@ -89,3 +89,37 @@
 		return
 	for(var/mob/M in viewers(5, location))
 		to_chat(M, notification)
+
+/proc/convert_pH_to_color(pH)
+	switch(pH)
+		if(14 to INFINITY)
+			return "#462c83"
+		if(13 to 14)
+			return "#63459b"
+		if(12 to 13)
+			return "#5a51a2"
+		if(11 to 12)
+			return "#3853a4"
+		if(10 to 11)
+			return "#3f93cf"
+		if(9 to 10)
+			return "#0bb9b7"
+		if(8 to 9)
+			return "#23b36e"
+		if(7 to 8)
+			return "#3aa651"
+		if(6 to 7)
+			return "#4cb849"
+		if(5 to 6)
+			return "#b5d335"
+		if(4 to 5)
+			return "#f7ec1e"
+		if(3 to 4)
+			return "#fbc314"
+		if(2 to 3)
+			return "#f26724"
+		if(1 to 2)
+			return "#ef1d26"
+		if(-INFINITY to 1)
+			return "#c6040c"
```

Comment:
```suggestion
#define COVERT_PH_TO_COLOR(pH, color) \
	switch(pH) {\
		if(14 to INFINITY)\
			{ color = "#462c83" }\
		if(13 to 14)\
			{ color = "#63459b" }\
		if(12 to 13)\
			{ color = "#5a51a2" }\
		if(11 to 12)\
			{ color = "#3853a4" }\
		if(10 to 11)\
			{ color = "#3f93cf" }\
		if(9 to 10)\
			{ color = "#0bb9b7" }\
		if(8 to 9)\
			{ color = "#23b36e" }\
		if(7 to 8)\
			{ color = "#3aa651" }\
		if(6 to 7)\
			{ color = "#4cb849" }\
		if(5 to 6)\
			{ color = "#b5d335" }\
		if(4 to 5)\
			{ color = "#f7ec1e" }\
		if(3 to 4)\
			{ color = "#fbc314" }\
		if(2 to 3)\
			{ color = "#f26724" }\
		if(1 to 2)\
			{ color = "#ef1d26" }\
		if(-INFINITY to 1)\
			{ color = "#c6040c" }\
		}
```
No need to use it if you think it's too cryptic, but macros are something good to know, and since they replace code at compile time they reduce proc overhead (in this situation it's not really an issue, since this proc is not called much, but for processing subsystems that make a lot of calls that can be more noticeable).
If you don't know what proc overhead is, well, on more sane languages there's zero-or-close-to-zero cost in calling a function/proc to execute code. Unfortunately that's not true here, and there's a small cost that can be noticed on hot code (not the case here).

As to how you'd use this, instead of doing `color = convert_pH_to_color(pH)` you'd do `COVERT_PH_TO_COLOR(pH, color)`

Reply:
I see! I'll try to implement this, seems easy enough